### ISSUE TITLE
php-parser undefined variable "c" (possible typo?)

### ISSUE THREAD

**Issue by nevadascout** [HvyIndustries/crane#151]
On line 757 of `lexer.js` there is a reference to a variable `c` which does not seem to exist. I assume this is a typo and should be `char` instead (as elsewhere in the code it is called with `char`)

Line in question:
**`757`** `|| IS_LABEL_START(c)`

Pull request opened here: https://github.com/glayzzle/php-parser/pull/18

```js
case 21:
  if (yy_.yytext == '\\') this.input();
  while(this._input.length > 0) {
    var char = this.input();
    if (char == '\\') {
      this.input();
    } else if ( char == '$' ) {
      if (
        this._input[0] == '{'
        || IS_LABEL_START(c) // <-- ########### HERE ###########
      ) {
        this.unput(char);
        break;
      }
    } else if (char == '`') {
      this.unput(char);
      break;
    }
  }
  // yy_.yytext = scan_escape_string(yy_.yytext);
  return T_ENCAPSED_AND_WHITESPACE;
```

_Framework: **FuelPHP 1.8**_

**Comment by nevadascout** [comment]
Update: changing `c` to `char` in the function call does indeed resolve this issue.

I will open a pull request on the php-parser project to fully resolve this.

**Comment by TheColorRed** [comment]
I wonder if the code has any other typos like that...

I could also be the cause of the `Undefined '0'` (forgot what the whole error says) issue.

**Comment by nevadascout** [comment]
If there is, I should find them at some point over the next hour or two -- I'm testing all the main php libraries/frameworks to see if they can be parsed properly

The `undefined 0` errors are thrown by the treebuilder trying to access a sub-array that doesn't exist. The root cause may well be the parser interpreting certain syntaxes or symbols incorrectly

**Comment by nevadascout** [comment]
Resolved in 5f807d5dac979e76fcccad1e7d0d835da30359b6
